File: crescer/las-vegas/api/src/main/java/br/com/cwi/crescer/api/security/config/SecurityConfig.java
package br.com.cwi.crescer.api.security.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpMethod;
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.session.web.http.HttpSessionIdResolver;

import static org.springframework.session.web.http.HeaderHttpSessionIdResolver.xAuthToken;

@Configuration
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
public class SecurityConfig extends WebSecurityConfigurerAdapter {

    @Bean
    public HttpSessionIdResolver httpSessionIdResolver() {
        return xAuthToken();
    }

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http
                .csrf().disable()
                .cors().and()

                .authorizeRequests().antMatchers("/*/**/publico").permitAll().and()

                .authorizeRequests().antMatchers(HttpMethod.POST, "/user/register").permitAll().and()

                .authorizeRequests().antMatchers(HttpMethod.POST, "/usuarios/{imageId}").permitAll().and()

                .authorizeRequests().anyRequest().authenticated()
                .and()
                .httpBasic();
    }
}

File: crescer/trabalhos/healthCare.api/src/main/kotlin/com/healthCare/api/Config/SecurityConfig.kt
@file:Suppress("DEPRECATION")

package com.healthCare.api.Config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpMethod
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.session.web.http.HeaderHttpSessionIdResolver.xAuthToken
import org.springframework.session.web.http.HttpSessionIdResolver


@Configuration
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
class SecurityConfig : WebSecurityConfigurerAdapter() {
    @Bean
    fun httpSessionIdResolver(): HttpSessionIdResolver {
        return xAuthToken()
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

    @Throws(Exception::class)
    override fun configure(http: HttpSecurity) {
        http
            .csrf().disable()
            .cors().and()
            .authorizeRequests()
            .antMatchers(HttpMethod.POST, "/user/create")
            .permitAll().and()
            .authorizeRequests()
            .anyRequest().authenticated()
            .and().httpBasic()
    }
}
